Data update

matthieu-nesme opened the pull request:
Adding a few utility functionalities to simplify Data tracking.

- externalizing Data tracking system outside of DataEngine so it can be use everywhere.
- Data tracking system is now using Data's counter (lighter and more simple).
- adding DataTrackerEngine that is a kind of DataEngine but that is not a component (not a BaseObject). It can easily be added anywhere and can link any Data, such as inside a Component, or even between components.

What you can do: (have a look at the tests)
- check if a Data changed on a regular rhythm. e.g check at each time step if a stiffness parameters changed to update a Matrix.
- automatically update a Data if its dependencies changed. e.g. young modulus can be set as a dependency of stiffness matrix, so each time you ask for the matrix, if the modulus changed, the matrix will be updated (behavior similar to DataEngine)

matthieu-nesme committed:
[dataupdate] let's begin!

matthieu-nesme committed:
[DataTracking] WIP
I do not really remember where I left off...

RomainTestylier committed:
Fix: remove explicit template in add skinning for support for both rigid and affine

RomainTestylier committed:
Fix: rigidScale API

RomainTestylier committed:
Add: list of skinned mesh in RigidScale python API

matthieu-nesme committed:
fixing warnings

matthieu-nesme committed:
BaseData: minor clean

RomainTestylier committed:
Add: uniform scale option for rigid scale
WIP: Rigid scale python API

matthieu-nesme committed:
[SofaKernel] cleaning MultiVec alloc/realloc

matthieu-nesme committed:
[Compliant] CompliantImplicitSolver: saving a few allocations with {pre|post}-stabilization

matthieu-nesme committed:
Merge remote-tracking branch 'upstream/master'

matthieu-nesme committed:
Merge remote-tracking branch 'upstream/master' into data-update

matthieu-nesme committed:
[data-update] implementing DataTrackerEngine, a kind of DataEngine but that is not a BaseObject. Its input and output Data can come from anywhere, and an external update function is given as a pointer.

matthieu-nesme committed:
Merge remote-tracking branch 'origin/data-update'

matthieu-nesme committed:
[SofaOpenglVisual] OglCylinderModel: adding an option to deactivate depth test.

matthieu-nesme committed:
[SofaOpenglVisual] OglCylinderModel: cleaning includes

matthieu-nesme committed:
[SofaOpenglVisual] OglCylinderModel: fixing depth test.

matthieu-nesme committed:
[SofaGeneralVisual] adding Visual3DText to display a camera-oriented (billboard) 3D text

(cherry picked from commit a09a7c648dcda6eddbcabbcfc49fb44f33a3aeee)

matthieu-nesme committed:
[SofaGeneralVisual] adding Visual3DText to display a camera-oriented (billboard) 3D text

matthieu-nesme committed:
[SofaGeneralVisual] Visual3DText: fixing depth test

matthieu-nesme committed:
Merge remote-tracking branch 'origin/Visual3DText'

# Conflicts:
#	modules/SofaGeneralVisual/Visual3DText.cpp

matthieu-nesme committed:
[SofaOpenglVisual] OglCylinderModel: fixing depth test.

matthieu-nesme committed:
[SofaGeneralVisual] Visual3DText: cleaning

matthieu-nesme committed:
drawTool: adding enable/disable depth test

matthieu-nesme committed:
glText: cleaning

matthieu-nesme committed:
VisualModel: fixing constness

matthieu-nesme committed:
VisualModel: adding function "hasTexture" that was only declared for OglModel. This info is needed for certain VisualManagers.

matthieu-nesme committed:
Visual3DText: cleaning

damienmarchal commented:
Thank Mathieu for this pull-request,

It seems interesting and at a first sight I would use your approach internally to a component to implement fine-grain update in the reinit function.

RomainTestylier committed:
Fix: compilation on Windows

RomainTestylier committed:
Fix: compilation on Windows

matthieu-nesme committed:
[SofaPython] it should fix the import of modules in the current directory

RomainTestylier committed:
Fix: remove debug messages

matthieu-nesme committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	modules/SofaGeneralVisual/Visual3DText.cpp
#	modules/SofaGeneralVisual/Visual3DText.h
#	modules/SofaOpenglVisual/OglCylinderModel.cpp

Commit:
[Flexible] some cleaning in quadrature engines

Commit:
[Flexible] add engines for strain smoothing, and examples

matthieu-nesme committed:
Merge remote-tracking branch 'upstream/master'

Commit:
[Flexible] fix templateName

matthieu-nesme committed:
Merge remote-tracking branch 'origin/master'

matthieu-nesme committed:
minor

matthieu-nesme committed:
Merge remote-tracking branch 'origin/master' into data-update

matthieu-nesme committed:
[Data-update] adding DataTrackerFunctor, a mechanism to trigger a functor as soon as a Data is modified.
Note this do not add any overcost to existing Data.

The basic idea:
DataTrackerFunctor is a DDGNode.
Its inputs are the tracked Data (there can be severals).
As soon as one of its input is modified, its setDirtyValue is called (regular Data graph behavior)
and can trigger its functor.

matthieu-nesme commented:
arg grosse bétise, j'ai pété la branche de la PR...
